src/pages/Auth/Register/index.js

```
@@ -7,71 +7,40 @@ import {
 } from "pages/Auth/components/styledForm";
 import { useInput } from "utils/hooks";
 import { setUsers } from "services/localStorage";
-import {
-  isEmpty,
-  isNotEmail,
-  isEmailInStorage,
-  isSamePassword,
-} from "utils/validators";
+import { isEmpty, isNotEmail, isSamePassword } from "utils/validators";
 
 const Register = () => {
-  const {
-    value: enteredName,
-    isTouched: nameIsTouched,
-    valueChangeHandler: nameChangeHandler,
-    valueBlurHandler: nameBlurHandler,
-    reset: resetNameInput,
-  } = useInput();
-  const {
-    value: enteredEmail,
-    isTouched: emailIsTouched,
-    valueChangeHandler: emailChangeHandler,
-    valueBlurHandler: emailBlurHandler,
-    reset: resetEmailInput,
-  } = useInput();
-  const {
-    value: enteredPassword,
-    isTouched: passwordIsTouched,
-    valueChangeHandler: passwordChangeHandler,
-    valueBlurHandler: passwordBlurHandler,
-    reset: resetPasswordInput,
-  } = useInput();
-  const {
-    value: enteredConfirmPass,
-    isTouched: confirmPassIsTouched,
-    valueChangeHandler: confirmPassChangeHandler,
-    valueBlurHandler: confirmPassBlurHandler,
-    reset: resetConfirmPassInput,
-  } = useInput();
+  const nameInputField = useInput();
+  const emailInputField = useInput();
+  const passwordInputField = useInput();
+  const confirmPassInputField = useInput();
```

send the validator on `useInput(validator)` so you can also return the errors upon the calling of the hook